core/model/modx/sources/modmediasource.class.php

```diff
@@ -811,45 +956,38 @@ public function renameObject($oldPath, $newName)
         $full_path = $fileObject->getPath();
 
         /* sanitize new path */
-        $newPath = $this->fileHandler->sanitizePath($newName);
-        $newPath = $this->fileHandler->postfixSlash($newPath);
-        $newPath = dirname($oldPath).'/'.$newPath;
+        $newPath = $this->sanitizePath($newName);
+        $newPath = $this->postfixSlash($newPath);
+        $newPath = dirname($oldPath) . '/' . $newPath;
 
         /* check to see if the new resource already exists */
         if ($this->filesystem->has($newPath)) {
             $this->addError('name', sprintf($this->xpdo->lexicon('file_err_ae'), $newName));
+
             return false;
         }
 
         if (!$this->checkFileType($newName)) {
-            // @TODO what should the error message be?
```

Comment:
was this todo resolved? Looks like it returns no error message now at all

Reply:
Yes - the error message is set in method `checkFileType`.